torch/csrc/lazy/core/dynamic_ir.cpp

```diff
@@ -0,0 +1,84 @@
+#pragma once
+
+#include <torch/csrc/lazy/core/dynamic_ir.h>
+
+namespace torch {
+namespace lazy {
+
+DimensionNode::DimensionNode(OpKind op, OpList operands, hash_t hash_seed):
+  Node(op, /*num_outputs=*/1, 
+  /* node_hash */ HashCombine(op.hash(), hash_seed),
+  /* dag_hash */ [&](bool bakeInSizes) { return OperandHashes(operands, HashCombine(op.hash(), hash_seed), bakeInSizes); }){
+    for (auto& operand : operands) {
+      if (!operand) {
+        continue;
+      }
+      // CHECK_LT(index, operand.node->num_outputs());
+      operands_.push_back(std::move(operand.node));
+      operands_as_outputs_.emplace_back(operands_.back().get(), operand.index);
+    }
+}
+
+std::string DimensionNode::ToString() const {
+  return "DimensionNode";
+}
+
+SizeNode::SizeNode(Value input, size_t dim): 
+    DimensionNode(OpKind{c10::Symbol::fromQualString("aten::size")}, {input}, MHash(dim)),
+    dim_(dim) {};
+
+int64_t SizeNode:: getStaticValue() const {
+    return dynamic_cast<const TsNode*>(operand(0).node)->shape(0).size(dim_);
```

Comment:
I have a feeling this wont work since you are extending torch::lazy::DimensionNode` from `torch::lazy::Node` rather than `TsNode`. I think it's okay to extend from `TsNode` for now or you could wait until Will lands this PR : https://github.com/pytorch/pytorch/pull/74071 . You might be able to land yr PR first.

Reply:
i landed #74071 FYI